core/arch/arm/mm/core_mmu.c

```diff
@@ -824,7 +864,8 @@ static void init_mem_map(struct tee_mmap_region *memory_map, size_t num_elems)
 		/* Map non-flat mapped addresses above flat mapped addresses */
 		va = ROUNDUP(va + CFG_TEE_RAM_VA_SIZE, CORE_MMU_PGDIR_SIZE);
```

Comment:
It is not very clean how `va` here is set to the end of the pager va space. Both "pager vaspace" end address and `va` here are computed using the same sequence, so that's fine now but not clean in the long term.
Maybe `add_pager_vaspace()` should update `end` so it can be reused here to define the virtual base address for the non-flat-mapped areas.